### PR TITLE
feat(clib): As a last resort, run `which sqlite3` to get library path

### LINKED ISSUE #168: Search `$PATH` for `sqlite3` or use `which sqlite3`
I'm on macOS and have sqlite installed via Nix package manager. Instead of searching a static list of paths, it should support alternative setups by searching `$PATH` or using `which sqlite3`.

### PR DESCRIPTION
**Pull request by mrjones2014**
Fixes #168 

As a last resort, if all other checks have run and the path still doesn't exist, check `which sqlite3`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ programs.neovim.plugins = [
 - https://github.com/nvim-telescope/telescope-frecency.nvim
 - https://github.com/kkharji/lispdocs.nvim
 - https://github.com/nvim-telescope/telescope-cheat.nvim
+- https://github.com/mrjones2014/legendary.nvim
```
Comment (mrjones2014) [inline README.md:76]: let me know if you'd like me to remove this and put it in a separate PR 🙂

Comment (kkharji) [reply README.md:76]: Feel like you already know the answer, Indeed that would be better 😬

File: lua/sqlite/defs.lua [hunk 2]
```diff
@@ -47,9 +47,33 @@ local clib = (function()
...
+      end
+
+      -- check for nix installation
+      path = string.format("%s/.nix_profile/bin/sqlite3", luv.os_getenv "HOME")
+      if file_exists(path) then
+        return path
+      end
+
+      -- as a last resort, try using `which sqlite3`
+      local handle = io.popen "which sqlite3"
```
[the remark this answers is not in the document]
Comment (mrjones2014) [reply lua/sqlite/defs.lua:65]: Hmm, it seemed to work for me. I'll do some more investigation today.

[the remark this answers is not in the document]
Comment (mrjones2014) [reply lua/sqlite/defs.lua:65]: It looks like there is actually not really a way to use `which sqlite3` to get the `dylib`. It looks like this worked for me on `nix` because the linked binary has the `dylib` statically in the binary or something. I'll close the PR.
